test/integration/services/modal.js

```diff
@@ -0,0 +1,182 @@
+
+    cy
+      .get('.modal--small')
+      .find('.modal-close')
+      .click()
+      .then(() => {
+        expect(buttonStub).to.be.calledOnce;
+        buttonStub.reset();
+      });
+
+    cy
+      .getRadio(Radio => {
+        const modal = Radio.request('modal', 'show:small', {
+          headingText: 'Close On Save',
+          bodyText: 'There\'s a callback to close the modal on save. Also a cool custom class!',
+          buttonClass: 'save-button',
+          className: 'modal--small custom-class',
+          onSubmit: () => {
+            modal.destroy();
+          },
+        });
+      });
+
+    cy
+      .get('.modal--small')
+      .then((modal) => {
+        expect(modal).to.have.class('custom-class');
+      })
+      .find('.save-button')
+      .click()
+      .should('not.exist');
+
+    cy
+      .getRadio(Radio => {
+        Radio.request('modal', 'show', {
+          footerView: false,
+          bodyText: 'No Footer',
+        });
+      });
+
+    cy
+      .get('.modal')
+      .then((modal) => {
+        expect(modal).to.not.have.class('modal-footer');
+      });
+
+    cy
+      .getRadio(Radio => {
+        const modal = Radio.request('modal', 'show', {
+          bodyText: 'Submit Disabled',
+          onSubmit: () => {
+            modal.disableSubmit(true);
+          },
+        });
+      });
+
+    cy
+      .get('.modal')
+      .find('.js-submit')
+      .click()
+      .should('be.disabled');
+
+    cy
+      .getRadio(Radio => {
+        Radio.request('modal', 'show', {
+          onSubmit: buttonStub,
+        });
+      });
+
+    cy
+      .get('.modal')
+      .find('.js-submit')
+      .click()
+      .then(() => {
+        expect(buttonStub).to.be.calledOnce;
+        buttonStub.reset();
+      });
+
+    cy
+      .getRadio(Radio => {
+        Radio.request('modal', 'show', {
+          onCancel: buttonStub,
+        });
+      });
+
+    cy
+      .get('.modal')
+      .find('.js-close')
+      .last() // Close Button
+      .click()
+      .then(() => {
+        expect(buttonStub).to.be.calledOnce;
+        buttonStub.reset();
+      });
+
+    smallOnTall();
+
+    // click the overlay, tall should stll be there
+    cy
+      .get('.fill-window--dark')
+      .last()
+      .click('left')
+      .then((fillWindow) => {
+        expect(fillWindow).to.not.have.class('modal--small');
+      })
+      .get('.modal--tall')
+      .contains('Tall Modal')
+      .get('.fill-window--dark')
+      .click('left')
+      .then((fillWindow) => {
+        expect(fillWindow).to.not.have.class('modal--tall');
+      });
+
+    cy
+      .getRadio(Radio => {
+        const modal = Radio.request('modal', 'show', {
+          bodyText: 'Hi',
+        });
+        modal.startPreloader();
+      });
+
+    cy
+      .get('.spinner')
+      .get('.fill-window--dark')
+      .click('right');
+
+    cy
+      .getRadio(Radio => {
+        Radio.request('modal', 'show', {
+          headerView: 'Custom Header',
```

Comment:
can we test that these worked?

Reply:
oh geez. I just left that hanging there didn't I